### PR TITLE
Deprecate collect_lua_garbage

### LINKED ISSUE #1814: Deprecate `collect_lua_garbage`
Since `collect_lua_garbage` in vshard 1.20.0 is deprecated, we should deprecate it too

### PR DESCRIPTION
**Pull request by yngvar-antonsson**
I didn't forget about

- [x] Tests
- [x] Changelog
- [x] Documentation

Close #1814

### CODE REVIEW

File: cartridge/webui/api-vshard.lua [hunk 1]
```diff
@@ -32,8 +32,8 @@ local gql_type_vsgroup = gql_types.object({
                 'in the storage group'
         },
         collect_lua_garbage = {
-            kind = gql_types.boolean.nonNull,
-            description = 'If set to true, the Lua collectgarbage() function is called periodically'
+            kind = gql_types.boolean,
+            description = 'Deprecated and has no effect anymore'
         },
```
Comment (olegrok) [inline cartridge/webui/api-vshard.lua:37]: I suggest to add "deprecationReason" as for collect_bucket_garbage_interval

File: cartridge/roles/vshard-router.lua [hunk 2]
```diff
@@ -161,6 +161,7 @@ local function bootstrap_group(group_name, vsgroup)
 
     local conf = confapplier.get_readonly()
     local vshard_cfg = vshard_utils.get_vshard_config(group_name, conf)
+    vshard_cfg.collect_lua_garbage = nil
```
Comment (olegrok) [inline cartridge/roles/vshard-router.lua:164]: Do you try to modify part of read-only config here?

File: cartridge/vshard-utils.lua [hunk 3]
```diff
@@ -520,7 +518,7 @@ local function get_vshard_config(group_name, conf)
         rebalancer_max_sending = vshard_groups[group_name].rebalancer_max_sending,
         sched_ref_quota = vshard_groups[group_name].sched_ref_quota,
         sched_move_quota = vshard_groups[group_name].sched_move_quota,
-        collect_lua_garbage = vshard_groups[group_name].collect_lua_garbage,
+        collect_lua_garbage = vshard_groups[group_name].collect_lua_garbage or false,
```
Comment (olegrok) [inline cartridge/vshard-utils.lua:521]: If user specifies `nil` here (or even omits at all) value shouldn't be converted to false

File: cartridge/roles/vshard-router.lua (hunk 2 above)
Comment (yngvar-antonsson) [reply cartridge/roles/vshard-router.lua:165]: `vshard_utils.get_vshard_config` returns a new table, not read only config

**Review by olegrok — APPROVED** [review]

**Pull request merged**